python-backend/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -0,0 +1,174 @@
+    @requires_role_mine_view
+    def get(self, mine_guid):
+        mine = Mine.find_by_mine_guid(mine_guid)
+        if not mine:
+            raise NotFound("Mine not found")
+        return mine.mine_incidents
+
+    @api.expect(parser)
+    @api.doc(description='creates a new incident for the mine')
+    @api.marshal_with(mine_incident_model, code=201)
+    @requires_role_mine_create
+    def post(self, mine_guid):
+        mine = Mine.find_by_mine_guid(mine_guid)
+        if not mine:
+            raise NotFound('Mine not found')
+
+        data = self.parser.parse_args()
+        incident = MineIncident.create(
+            mine,
+            data['incident_timestamp'],
+            data['incident_description'],
+            followup_type_code=data['followup_type_code'] or 'UND',
+            followup_inspection_no=data['followup_inspection_no'],
+            reported_timestamp=data['reported_timestamp'],
+            reported_by=data['reported_by'],
+            reported_by_role=data['reported_by_role'],
+        )
+        incident.save()
+
+        return incident, 201
+
+
+class MineIncidentResource(Resource, UserMixin):
+    parser = reqparse.RequestParser(trim=True)
+    parser.add_argument(
+        'incident_timestamp',
+        help='Datetime of when the incident occured ',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'incident_description',
+        help='reported details of the incident',
+        type=str,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'reported_timestamp',
+        help='Datetime of when the incident was reported',
+        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'reported_by',
+        help='Name of party who reported the incident',
+        type=str,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'reported_by_role',
+        help='Job title of incident reporter',
+        type=str,
+        location='json',
+        store_missing=False)
+    parser.add_argument(
+        'followup_type_code',
+        help='Mark incident to have a follow up inspection',
+        location='json',
+        type=str,
+        store_missing=False)
+    parser.add_argument(
+        'followup_inspection_no',
+        help='NRIS inspection related to this incident',
+        location='json',
+        type=str,
+        store_missing=False)
+    parser.add_argument(
+        'closing_report_summary',
+        help='Report from mine in reaction to the incident',
+        location='json',
+        type=str,
+        store_missing=False)
+
+    @api.marshal_with(mine_incident_model, code=200)
+    @requires_role_mine_view
+    def get(self, mine_incident_guid, mine_guid=None):
```

Comment:
If the mine_guid is not used can you please remove it.

Reply:
Yea there were two paths to get here, but we discussed and agreed there should only be one.